src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -1,12 +1,105 @@
+                                                  Organization performingOrganization) {
+        DiagnosticReport fhir = new DiagnosticReport();
+        fhir.setId(uuidGenerator.generateUUID());
+
+        ServiceReportDetails serviceReportDetails = message.getServiceReportDetails();
+
+        // fhir.issued = SG2.DTM
+        mapIssued(serviceReportDetails.getDateIssued(), fhir);
+        // fhir.status
+        fhir.setStatus(STATUS_MAP.get(serviceReportDetails.getStatus().getEvent()));
+        // fhir.identifier
+        mapIdentifier(serviceReportDetails.getReference(), fhir);
+        // fhir.code
+        mapCode(fhir);
+        // fhir.subject
+        fhir.getSubject().setReference(fullUrlGenerator.generate(patient));
+        // fhir.specimens
+        mapSpecimens(specimens, fhir);
+        // fhir.results
+        mapObservations(observations, fhir);
+        // fhir.performer
+        mapPerformer(performingPractitioner, performingOrganization, fhir);
+        /* TODO: Add BasedOn - ProcedureReport & Result - Observation */
+        return fhir;
+    }
+
+    private void mapIssued(final DiagnosticReportDateIssued reportIssuedDate, final DiagnosticReport fhir) {
+        LocalDateTime dateIssued = reportIssuedDate.getDateIssued();
+        ZonedDateTime zonedDateTime = dateIssued.atZone(TimestampService.UK_ZONE);
+        fhir.setIssued(Date.from(zonedDateTime.toInstant()));
+    }
+
+    private void mapIdentifier(final Reference reference, final DiagnosticReport fhir) {
+        final Identifier identifier = new Identifier();
+        identifier.setValue(reference.getNumber());
+        fhir.addIdentifier(identifier);
+    }
+
+    private void mapSpecimens(final List<Specimen> specimens, final DiagnosticReport fhir) {
+        Optional.ofNullable(specimens).ifPresent(
+            s -> s.forEach(specimen -> fhir.addSpecimen().setReference(fullUrlGenerator.generate(specimen))));
+    }
+
+    private void mapCode(final DiagnosticReport fhir) {
+        fhir.getCode().addCoding()
+            .setDisplay(CODE_DISPLAY)
+            .setCode(CODE_NUMBER)
+            .setSystem(CODE_SYSTEM);
+    }
+
+    private void mapPerformer(Practitioner performingPractitioner, Organization performingOrganization,
+                              final DiagnosticReport fhir) {
+        Optional.ofNullable(performingOrganization).ifPresent(organization ->
+            fhir.addPerformer().getActor().setReference(fullUrlGenerator.generate(organization)));
+        Optional.ofNullable(performingPractitioner).ifPresent(performer ->
+            fhir.addPerformer().getActor().setReference(fullUrlGenerator.generate(performer)));
+    }
+
+    private void mapObservations(final List<Observation> observations, final DiagnosticReport fhir) {
+        Optional.ofNullable(observations).ifPresent(
+            obs -> obs.forEach(observation -> fhir.addResult().setReference(fullUrlGenerator.generate(observation))));
```

Comment:
The `List<Observations>` should never be `null`; if there are none then it will be the empty list.
```suggestion
        observations.stream()
            .map(fullUrlGenerator::generate)
            .forEach(reference -> fhir.addResult().setReference(reference));
```

Reply:
Removed the unnecessary null check in 9a98aa8